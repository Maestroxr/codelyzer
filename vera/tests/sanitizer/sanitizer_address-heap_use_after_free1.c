#include<stdio.h> 
#include<stdlib.h> 
int main(void) 
{ 
	char* p = (char*)malloc(3); 
	if (!p)
	{
		printf("Memory error");
	}
	*p = 'a'; 
	*(p+1) = 'b'; 
	*(p+2) = 'c'; 
	free(p); 
	*p = 'a'; 
	return 0; 
}
