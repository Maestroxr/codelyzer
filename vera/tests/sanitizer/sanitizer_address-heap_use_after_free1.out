C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_use_after_free1.c:14: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_use_after_free1] Heap use after free, suspect variable: 'p', sanitizer_address-heap_use_after_free1.c:14
	Was accessed by write of size 1 at:
		sanitizer_address-heap_use_after_free1.c:14 in function 'main'
	Accessed address is located 0 bytes inside of 3-byte region, freed by thread T0 here:
		sanitizer_address-heap_use_after_free1.c:13 in function 'main'
	Previously allocated by thread T0 here:
		sanitizer_address-heap_use_after_free1.c:5 in function 'main'
:MemoryEnd:
